_pages/dev/performance/core-web-vitals/prioritize-largest-content-paint-image.md

```diff
@@ -0,0 +1,141 @@
+The selected OOTB Spartacus components implementations that currently can be controlled with `lcpCmsComponents` are:
+
+- `BannerComponent`
+- `ProductCarouselComponent`
+- `ProductImageComponent` (and its related `ProductImageZoomProductImagesComponent`)
+  (if you need it in other components, you can implement a custom component or request it as a new feature in OOTB Spartacus)
+
+### Example of configuring `lcpCmsComponents`
+To give an example, supposing I've analyzed my pages in my storefront and I know that the LCP images in the following types of pages are:
+
+- on Homepage: it's CMS component with ID `"ElectronicsHomepageSplashBannerComponent"`
+- on Product Details Page: it's CMS component with ID `"ProductImagesComponent"`
+- on Product Listing Page: it's CMS component with ID `"ProductListComponent"`
+
+... then I can configure the `lcpCmsComponents` in my Spartacus app like this:
+
+Option 1: static list of CMS component IDs:
+
+```typescript
+provideConfig({
+  lcpCmsComponents: {
+    ids: [
+      'ElectronicsHomepageSplashBannerComponent',
+      'ProductImagesComponent',
+      'ProductListComponent',
+    ],
+  },
+}),
+```
+
+Option 2: Change the CMS component IDs in the CMS data to include the special marker `__cxLCP__`, for example:
+
+```
+ElectronicsHomepageSplashBannerComponent__cxLCP__
+ProductImagesComponent__cxLCP__
+ProductListComponent__cxLCP__
+```
+
+Then you don't need to configure Spartacus `lcpCmsComponents.ids` at all, as Spartacus will automatically recognize the components with the marker in their IDs (default marker is `__cxLCP__`).
+
+If you want to change the default marker, you can do it like this:
+
+```typescript
+provideConfig({
+  lcpCmsComponents: {
+    idMarker: '--myCustomLcpMarker--,
+  },
+}),
+```
+
+### Implementation details of components supporting `lcpCmsComponents`
+
+The components that support the `lcpCmsComponents` configuration will automatically set the input `[fetchPriority]="ImageFetchPriority.HIGH"` on the `<cx-media>` component inside them, if the CMS component is recognized as containing the Largest Contentful Paint image. They take this information from the Spartacus directive  `*cxLcpContext` in the HTML template of the component. Alternatively, they can inject the `LCP_CONTEXT` injection token in their typescript code.
+For example, the `BannerComponent` HTML template looks like this:
+
+```html
+<ng-container *cxLcpContext="let lcpContext">
+  <cx-media
+    [fetchPriority]="lcpContext.fetchPriority$ | async"
+    ...other-inputs-here...
+  ></cx-media>
+</ng-container>
+```
+
+Note: If the CMS component contains multiple `<cx-media>` components (e.g. main image and a few thumbnails), the input `[fetchPriority]="ImageFetchPriority.HIGH"` should on the `<cx-media>` component that contains the Largest Contentful Paint image, and not on all of them. Otherwise all those images to be loaded eagerly with high priority, which is not recommended.
```

Comment:
```suggestion
Note: If the CMS component contains multiple `<cx-media>` components (e.g. main image and a few thumbnails), the input `[fetchPriority]="ImageFetchPriority.HIGH"` should be on the `<cx-media>` component that contains the Largest Contentful Paint image, and not on all of them. Otherwise all those images will be loaded eagerly with high priority, which is not recommended.
```